mautic/mautic
# Fix for issue #5103 - Company/Accounts being duplicated on CRM Sync o…

…f edited Company/Account

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? |  Yes
| New feature? |  No
| Automated tests included? | No
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) |  #5103 
| BC breaks? |  No
| Deprecations? |  No

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description: 
Like the reporter of #5103 we were seeing duplicate Mautic Company records being created when a SuiteCRM Account (Company) was edited. After debugging through this it was discovered that inside IdentifyCompanyHelper::identifyLeadsCompany class and method there was an else if statement for looking up a company name (not domain) through the MX records of the companyemail field. If anything was in that field it correctly skipped the next else if statement where the existing Mautic company would have been looked up by the 'companyname' field. 

The proper ordering should be to first lookup my the name then companyname fields. I believe that the else if check put in to lookup the domain should be assigned to the $companyDomain variable instead of the $companyName variable and have made that adjustment in this pull request as well.  Making these adjustments corrected the issue when testing in my local environment. Please let me know if you have any questions or feedback.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Activate SugarCRM integration (6.x) and map any applicable fields for account (companyname and companyemail at a minimum)
2. Add a SugarCRM Account (company) record and trigger the `mautic:integration:fetchleads --integration=Sugarcrm --time-interval="1 minute"` command
3. View that account/company added in Mautic
4. Edit the same SugarCRM Account (company) just added and trigger the `mautic:integration:fetchleads --integration=Sugarcrm --time-interval="1 minute"` command again
5. Observe that a duplicate Mautic Company is created versus updating the existing one.

#### Steps to test this PR:
1. First test the bug using the steps above
2. Pull the included commit into your repo
3. Follow the same testing steps above
4. Observe that the existing Mautic Company is properly updated versus a duplicate record being created

#### List deprecations along with the new alternative:
1. N/A

#### List backwards compatibility breaks:
1. None that I am aware of
